Framework/API/src/ISpectrum.cpp

```diff
@@ -135,6 +135,22 @@ bool ISpectrum::hasDx() const { return bool(histogramRef().sharedDx()); }
  */
 void ISpectrum::resetHasDx() { mutableHistogramRef().setSharedDx(nullptr); }
 
+/**
+ * Makes sure the histogram has valid Y data.
+ * @param histogram A histogram to process.
+ * @throw std::invalid_argument if Y data is NULL.
+ */
+void ISpectrum::checkAndSanitizeHistogram(HistogramData::Histogram &histogram) {
```

Comment:
Semantically, it would probably make more sense to have this implementation in `Histogram1D`, since `EventList` (which also inherits `ISpectrum`) has a different implementation. Having this in the common base class is a bit confusing.